ethexe/processor/src/handling/run.rs

```diff
@@ -171,24 +171,22 @@ pub async fn run(
                     state.hash = new_state_hash;
                 });
 
-                if !program_journals.is_empty() {
-                    for (journal, dispatch_origin) in program_journals {
-                        let mut journal_handler = JournalHandler {
-                            program_id,
-                            dispatch_origin,
-                            controller: TransitionController {
-                                transitions: in_block_transitions,
-                                storage: &db,
-                            },
-                        };
-                        core_processor::handle_journal(journal, &mut journal_handler);
-                    }
+                if program_journals.is_empty() {
+                    continue;
                 }
```

Comment:
I just noticed the branch can be just omitted because the following `for`-loop is not going to be executed anyway

Reply:
Fixed